relayer/checkpoint/checkpoint.go

```diff
@@ -77,7 +78,8 @@ func (cm *CheckpointManager) writeToDatabase() {
 	cm.lock.RLock()
 	defer cm.lock.RUnlock()
 	// Defensively ensure we're not writing the default value
-	if cm.committedHeight == 0 {
```

Comment:
I think we should change the lock acquisition above to just regular lock/unlock since we're writing to the `dirty` variable now.